### PR TITLE
BWS31: + common component (button, input, checkbox, select)

### PR DESCRIPTION
**Pull request by Zamay**

### CODE REVIEW

**Comment by Zamay** [comment]
photo_2022-02-08 13 25 31

File: frontend/src/components/common/select/select.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        //     background: '#393939',
+        //     borderRadius: '50px',
+        //     border: '2px solid #8F8F8F',
+        //     height: '45px',
+        //     cursor: 'pointer' }),
+        //   menu: (provided) => ({
+        //     ...provided,
+        //     background: '#393939',
+        //     cursor: 'pointer' }),
+        //   singleValue: (provided) => ({ ...provided, color: '#fff' }),
```
Comment (tbezrukava) [inline frontend/src/components/common/select/select.tsx]: creteria [B20](https://github.com/BinaryStudioAcademy/quality-criteria/blob/production/source/javascript.md)

File: frontend/package.json [hunk 2]
```diff
@@ -24,7 +24,8 @@
     "react-hook-form": "7.24.0",
     "react-redux": "7.2.6",
     "react-router-dom": "6.2.1",
-    "react-scripts": "5.0.0"
+    "react-scripts": "5.0.0",
+    "react-select": "^5.2.2"
```
Comment (tbezrukava) [inline frontend/package.json]:
```suggestion
    "react-select": "5.2.2"
```
creteria [B19](https://github.com/BinaryStudioAcademy/quality-criteria/blob/production/source/javascript.md)

File: frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { Button } from 'components/common/common';
+// import { Button } from 'components/common/common';
```
Comment (what1s1ove) [inline frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx]: remove it

File: frontend/src/components/common/button/button.tsx [hunk 4]
```diff
@@ -1,12 +1,27 @@
-import { ButtonType } from 'common/enums/enums';
+import { ButtonType, ButtonColor } from 'common/enums/enums';
+import clsx from 'clsx';
+import styles from './styles.module.scss';
 
 type Props = {
   label: string;
+  onClick: React.MouseEventHandler<HTMLButtonElement>;
   type?: ButtonType;
+  color?: ButtonColor;
```
Comment (what1s1ove) [inline frontend/src/components/common/button/button.tsx]:
```suggestion
  btnStyle?: ButtonStyle;
```

File: frontend/src/common/enums/ui/button-color.enum.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+enum ButtonColor {
+  ORANGE = 'orange',
+  BLACK = 'black',
```
Comment (what1s1ove) [inline frontend/src/common/enums/ui/button-color.enum.ts]:
```suggestion
enum ButtonStyle {
  FILLED = 'filled',
  OUTLINED = 'outlined',
```

File: frontend/src/components/common/button/styles.module.scss [hunk 6]
```diff
@@ -0,0 +1,19 @@
+.btn {
+  width: 100%;
+  height: 45px;
+  margin: 10px 0;
+  color: white;
+  font-weight: 800;
+  font-size: 16px;
+  border: 2px solid #ff9c00;
+  border-radius: 50px;
+  cursor: pointer;
```
Comment (what1s1ove) [inline frontend/src/components/common/button/styles.module.scss]:
```suggestion
```
button has it as default

File: frontend/src/components/common/button/styles.module.scss [hunk 7]
```diff
@@ -0,0 +1,19 @@
+.btn {
+  width: 100%;
+  height: 45px;
```
Comment (what1s1ove) [inline frontend/src/components/common/button/styles.module.scss]:
```suggestion
  min-height: 45px;
```

File: frontend/src/components/common/button/styles.module.scss [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+  color: white;
+  font-weight: 800;
+  font-size: 16px;
+  border: 2px solid #ff9c00;
+  border-radius: 50px;
+  cursor: pointer;
+}
+
+.btn-orange {
+  background-color: #ff9c00;
```
Comment (what1s1ove) [inline frontend/src/components/common/button/styles.module.scss]: add custom properties for colours `frontend/src/assets/css/variables.scss`

File: frontend/src/components/common/input/input.tsx [hunk 9]
```diff
@@ -6,6 +6,8 @@ import {
 import { useFormControl } from 'hooks/hooks';
 import { InputType } from 'common/enums/enums';
 import { ErrorMessage } from 'components/common/common';
+import clsx from 'clsx';
```
Comment (what1s1ove) [inline frontend/src/components/common/input/input.tsx]: use `getValidClasses`

File: frontend/src/components/common/input/styles.module.scss [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+.input:not([value=""]):not(:invalid) {
+  border: 2px solid #ff9c00;
+}
+
+.input-error {
+  border: 2px solid #ff0000;
+}
+
+label {
```
Comment (what1s1ove) [inline frontend/src/components/common/input/styles.module.scss]: don't use global styles

File: frontend/src/components/common/select/styles.module.scss [hunk 11]
```diff
@@ -0,0 +1,22 @@
+.react-select__container {
```
Comment (what1s1ove) [inline frontend/src/components/common/select/styles.module.scss]:
```suggestion
.reactSelectContainer {
```
use kebab case

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: .env.example/backend.env [hunk 12]
```diff
@@ -6,9 +6,6 @@ PORT=3001
 HOST=0.0.0.0
 FRONTEND_URL=http://localhost:3000
 #
-# AUTHENTICATION
-SECRET_KEY={ENTER YOUR SECRET KEY HERE}
```
Comment (what1s1ove) [inline .env.example/backend.env:10]: you delete a lot of think
left them. you should bring/change only change that you need in your task

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Pull request closed without merging**